rskj-core/src/main/java/co/rsk/net/MessageCounter.java

```diff
@@ -0,0 +1,46 @@
+package co.rsk.net;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Keeps the counter of messages per peer
+ */
+public class MessageCounter {
+
+	private static final AtomicInteger ZERO = new AtomicInteger(0);
+
+	private Map<NodeID, AtomicInteger> messagesPerNode = new ConcurrentHashMap<>();			
+    
+    public int getValue(Peer sender) {
+    	return Optional.ofNullable(messagesPerNode.get(sender.getPeerNodeID())).orElse(ZERO).intValue();
```

Comment:
would add some parentheses so it is easy to read on the fly (and .intValue() would be outside the parentheses).